src/GSM.cpp

```diff
@@ -100,12 +100,14 @@ int GSM::isAccessAlive()
 
 bool GSM::shutdown()
 {
-  MODEM.send("AT+CPWROFF");
+  if (_state == READY_STATE_DONE) {
+    MODEM.send("AT+CPWROFF");
 
-  if (MODEM.waitForResponse(40000) == 1) {
-    MODEM.end();
-    _state = GSM_OFF;
-    return true;
+    if (MODEM.waitForResponse(40000) == 1) {
+      MODEM.end();
+      _state = GSM_OFF;
+      return true;
+    }
```

Comment:
I think `MODEM.end()` and the state reset should still be called if the state is not done. What do you think?

Reply:
ummm yes in this way the reset is forced any way like the secureShutdown, but in this case the ```MODEM.waitForResponse(40000)``` should be removed i think

Reply:
changed @sandeepmistry